src/main.rs

```diff
@@ -484,6 +484,42 @@ impl Config {
     }
 }
 
+/// Translates a tag-like bound (such as `1.62.0`) to a `Bound::Date` so that
+/// bisecting works for versions older than 167 days.
+fn fixup_bounds(
+    access: &Access,
+    start: &mut Option<Bound>,
+    end: &mut Option<Bound>,
+) -> anyhow::Result<()> {
+    let is_tag = |bound: &Option<Bound>| -> bool {
+        match bound {
+            Some(Bound::Commit(commit)) => commit.contains('.'),
+            None | Some(Bound::Date(_)) => false,
+        }
+    };
+    let is_datelike = |bound: &Option<Bound>| -> bool {
+        matches!(bound, None | Some(Bound::Date(_))) || is_tag(bound)
+    };
+    if !(is_datelike(start) && is_datelike(end)) {
+        // If the user specified an actual commit for one bound, then don't
+        // even try to convert the other bound to a date.
+        return Ok(());
+    }
+    let fixup = |bound: &mut Option<Bound>| -> anyhow::Result<()> {
+        if is_tag(bound) {
+            if let Some(b) = bound.clone() {
+                if is_tag(bound) {
+                    *bound = Some(Bound::Date(access.repo().bound_to_date(b)?));
```

Comment:
Maybe print a message here saying that this happened. That seems like a good thing for the user to know about.

Reply:
Sounds good!